Third add

muja-b opened the pull request.

muja-b committed:
fourth add added

muja-b committed:
last Add impemented

sudqijawabreh commented:
Consider using .gitignore files to ignore generated files like dlls

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+        private int tryAddSimple(string Numbers)
         {
             int ans;
             ans = Numbers.Where(x => x != ',').Aggregate((sum, i) => sum += i);
```
This is an over engineering with a lot of hacks😄

sudqijawabreh replied:
over engineering because it works for more than 2 numbers

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
         }
+        private int tryAddbig(string Numbers)
+        {
+            List<string> Nums = Numbers.Split(',').ToList();
```
this is better than the first one

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+            int sum = GetSum(Nums);
+            return sum;
+        }
         private int tryAddWithEndLine(string Numbers)
```
wrong method name made it misleading 
withNewLine not end line

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
-            ans = Numbers.Where(x => x != ',').Aggregate((sum, i) => sum += i);
-            ans -= Numbers.Where(x => x != ',').Count() * '0';
-            return ans;
+            Numbers = Numbers.Replace("/n", ",");
```
mutating the input is not a very good idea.

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+            {
+                diameter = Getdiameter(Numbers);
+            }
+            List<string> Nums = Numbers.Split(diameter).ToList();
```
the requirement says all old cases should still be supported 
I don't think the comma case is still supported here.

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
 
+        private void RemoveNewLine(string numbers, char diameter)
+        {
+            numbers = numbers.Replace("/n", diameter + "");
```
maybe you could just have used .ToString

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+        public char Getdiameter(string numbers)
+        {
+            char diameter = numbers[2];
+            numbers = numbers.Remove(0, 3);
```
mutating the input is not a good idea here too.

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+            numbers = numbers.Replace("/n", diameter + "");
+        }
+
+        public char Getdiameter(string numbers)
```
typo delimiter

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+            }
+            RemoveNewLine(Numbers, diameter);
+            List<string> Nums = Numbers.Split(diameter).ToList();
+            testNegitaves(Nums);
```
use more clearer names for the methods

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+                if (nums.ElementAt(i).Count() > 3)
+                {
+                    var LargeNum = nums.ElementAt(i).Substring(nums.ElementAt(i).Count()-3,3);
+                    nums.RemoveAt(i);
```
again mutation is not a good idea.

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+        {
+            for (int i = 0; i < nums.Count(); i++)
+            {
+                if (nums.ElementAt(i).Count() > 3)
```
this is a hacky why 
consider the users says bigger than 4500 then you need to change the whole method
and the filter method doesn't clearly show what is doing 
you need to run the code or read through it carefully to figure out the condition that numbers should be less than 1000

sudqijawabreh replied:
also why the use of ElementAt 
couldn't we just use [] operator ?

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+            {
+                if (nums.ElementAt(i).Count() > 3)
+                {
+                    var LargeNum = nums.ElementAt(i).Substring(nums.ElementAt(i).Count()-3,3);
```
why we are doing the substring I don't get it

sudqijawabreh commented on Program.cs:
```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
+            }
+        }
+
+        private int GetSum(List<string> nums)
```
sum is doing more than one thing

sudqijawabreh reviewed (commented):
good job

muja-b committed:
Separated the Getsum into two functions

Getsum used to parse and add the items of an array
now I added a function to parse separately

muja-b committed:
Fixed the mutations around the int list
